specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+  /// from least to most stable. If a `RuntimeChannelPreferenceSet` has been provided, the
+  /// loader will only search for channels in the set. See `COREWEBVIEW2_RUNTIME_CHANNEL`
+  /// for more details on channels. This property can be overridden by the corresponding
+  /// registry key `ReverseChannelSearchOrder` or the environment variable
+  /// `WEBVIEW2_REVERSE_CHANNEL_SEARCH_ORDER`. Set the value to `1` to reverse the search
+  /// order. See `CreateCoreWebView2EnvironmentWithOptions` for more details on
+  /// overrides.
+  [propget] HRESULT ShouldReverseChannelSearchOrder([out, retval] BOOL* value);
+
+  /// Sets the `ShouldReverseChannelSearchOrder` property.
+  [propput] HRESULT ShouldReverseChannelSearchOrder([in] BOOL value);
+}
+
+/// Get the browser version info including channel name if it is not the
+/// WebView2 Runtime.  Channel names are Beta, Dev, and Canary.
+/// If an override exists for `browserExecutableFolder`, `runtimeChannelPreferenceSet`,
+/// or `reverseChannelSearchOrder`, the override is used.  `browserExecutableFolder`
+/// takes precedence over the other options. See `CreateCoreWebView2EnvironmentWithOptions`
+/// for more details on overrides. If an override is not specified, then the
+/// parameters passed to `GetAvailableCoreWebView2BrowserVersionStringWithOptions`
+/// are used. Returns `HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)` if it fails to find
+/// an installed WebView2 Runtime or non-stable Microsoft Edge installation.
+///
+/// The caller must free the returned string with `CoTaskMemFree`.  See
+/// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+STDAPI GetAvailableCoreWebView2BrowserVersionStringWithOptions(
+    PCWSTR browserExecutableFolder,
+    ICoreWebView2EnvironmentOptions* environmentOptions,
+    LPWSTR* versionInfo);
+```
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    // The WebView2 Runtime channel. Use `RuntimeChannelPreferenceSet` and
+    // `ShouldReverseChannelSearchOrder` on `CoreWebView2EnvironmentOptions` to
+    // control which channel the WebView2 loader searches for.
+    // |Channel|Primary purpose|How often updated with new features|
+    // |:---:|---|:---:|
+    // |Stable (WebView2 Runtime)|Broad Deployment|~4 weeks|
+    // |Beta|Flighting with inner rings, automated testing|~4 weeks|
+    // |Dev|Automated testing, selfhosting to test new APIs and features|Weekly|
+    // |Canary|Automated testing, selfhosting to test new APIs and features|Daily|
+    [Flags] enum CoreWebView2RuntimeChannel
+    {
+        // The stable WebView2 Runtime that is released every 4 weeks.
+        Stable = 1,
+        // The Beta Runtime channel that is released every 4 weeks, a week before the stable release.
+        Beta = 2,
+        // The Dev Runtime channel that is released weekly.
+        Dev = 4,
+        // The Canary Runtime channel that is released daily.
+        Canary = 8,
```

Comment:
Consider adding overloads of the version strings where we pull the channel out of the string as one of these enums